test/e2e/util/util.go

```diff
@@ -283,3 +290,144 @@ func EnsureNodeCountMatchesNodePoolReplicas(t *testing.T, ctx context.Context, h
 		}
 	})
 }
+
+func EnsureAPIBudget(t *testing.T, ctx context.Context, client crclient.Client, hostedCluster *hyperv1.HostedCluster) {
+	t.Run("EnsureAPIBudget", func(t *testing.T) {
+
+		// Get hypershift-operator token
+		operatorServiceAccount := &corev1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "operator",
+				Namespace: "hypershift",
+			},
+		}
+		if err := client.Get(ctx, crclient.ObjectKeyFromObject(operatorServiceAccount), operatorServiceAccount); err != nil {
+			t.Fatalf("failed to get hypershift operator service account: %v", err)
+		}
+		var secretName string
+		for _, secret := range operatorServiceAccount.Secrets {
+			if strings.HasPrefix(secret.Name, "operator-token-") {
+				secretName = secret.Name
+				break
+			}
+		}
+		if secretName == "" {
+			t.Fatal("failed to determine hypershift operator token secret name")
+		}
+		tokenSecret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      secretName,
+				Namespace: "hypershift",
+			},
+		}
+		if err := client.Get(ctx, crclient.ObjectKeyFromObject(tokenSecret), tokenSecret); err != nil {
+			t.Fatalf("failed to get hypershift operator token secret: %v", err)
+		}
+		token, ok := tokenSecret.Data["token"]
+		if !ok {
+			t.Fatal("token secret did not contain a token value")
+		}
+
+		// Get thanos-querier endpoint
+		promRoute := &routev1.Route{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "thanos-querier",
+				Namespace: "openshift-monitoring",
+			},
+		}
+		if err := client.Get(ctx, crclient.ObjectKeyFromObject(promRoute), promRoute); err != nil {
+			t.Skip("unable to get prometheus route, skipping")
+		}
+		promEndpoint := fmt.Sprintf("https://%s", promRoute.Status.Ingress[0].Host)
```

Comment:
Shouldn't happen but maybe lencheck this first so we don't cause an out of bounds if for some reason the route doesn't have a host?